pyproject.toml

```diff
@@ -85,9 +85,7 @@ known_first_party = "poetry.core"
 known_third_party = ["poetry.core._vendor"]
 
 [tool.mypy]
-python_version = "3.6"
```

Comment:
I'd like to keep this, as I understand it will matter for 3.7's `from __future__ import annotations` and similar features. In general with our linting tools I think it's a good idea to call out the minimum Python version.

Reply:
i don't think this configuration setting sets the minimum version for the syntax, just the interpreter version used to run mypy, which is not *quite* the same thing.

Can you explain to me the interaction with `from __future__ import annotations`? i'm not sure I understand that. it won't impact us until the move to 3.7+, and even though i'm not sure I understand why it's an issue?

My impression was-

> In general with our linting tools I think it's a good idea to call out the minimum Python version

I'm not sure i agree with this. the approach i would usually take is to

1. test with the oldest version I can get away with
2. lint with the newest version I can get away with (without breaking 1.!)

Reply:
Fair enough, looks like I got misled.